float32 cx
float32 cy
float32 area
float32 ux
float32 uy
float32 vx
float32 vy
